src/modules/roc_pipeline/receiver_session_group.cpp

```diff
@@ -64,6 +66,10 @@ size_t ReceiverSessionGroup::num_sessions() const {
 
 bool ReceiverSessionGroup::can_create_session_(const packet::PacketPtr& packet) {
     if (packet->flags() & packet::Packet::FlagRepair) {
+        while (packet_ring_buffer_.size() >= packet_ring_buffer_size_) {
+            packet_ring_buffer_.remove(packet_ring_buffer_.front());
+        }
+        packet_ring_buffer_.push_back(packet);
```

Comment:
Let's extract this piece into a method and move its call to route_packet(). This way the high-level logic will be clear from route_packet body: try to route packet to existing session, then try to create a new one, finally add it to the ring buffer.